### ISSUE TITLE
Post comments to PRs as Slack Thread replies

### ISSUE THREAD

**Issue by timexcession** [integrations/slack#967]
It would be neat if the PR integration would create and manage slack threads.

We have a pretty busy repo, with many PRs on going at once - it would be great if each PR could be managed through a thread on slack rather than having to search and filter an entire channel.

This relates to: https://github.com/integrations/slack/issues/660

Thanks for considering!

**Comment by Startouf** [comment]
On a similar topic, it would be great if PR reviews (which seem to be considered separately than PR comments) could also be sent to the thread started by the `pulls` notification "Pull request opened by x" if it exists.

The difficulty of this feature (both for reviews and comments) is finding the original message(s). 

Right now it's possible to subscribe to PR notifications (+ comments/reviews) in any slack channel, so the code would require to try to find all the non-threaded messages in each channel the notification should be sent to, if they exist (because it's possible to subscribe to PR notifications/comments even if the original PR message was not sent to Slack).

My suggestion to handle PR reviews and comments in thread (I'm not sure about the feasibility of the search part)

1. Change the original PR notification message to include the issue number and the repo name, so we can find it much more easily when searching messages.

```
GitHubAPP  3:15 PM
Pull request opened by X
 + attachment
```

Should become

```
GitHubAPP  3:15 PM
Pull request #{PR_NUMBER} opened in {repo_name} by X
 + attachment
```

2. Before posting a comment/review, search in slack, in each channel the subscription is active, to find a message containing exactly `Pull request #{PR_NUMBER} opened in {repo_name}`. Take the last matching message in each channel (by created date), its ID will become the relevant thread_id of the channel. 

I'm not sure this type of search is supported/it would require extra permissions

3. in each channel the subscription is active, either we have found a thread_id or we have not => either send the message threaded under the message with thread_id or send the notification as a new message

**Comment by PatrikSeal** [comment]
The fact that pull-request has so much slack channel space caused my team to disable it.  We are jus pasting the PR link when we want a review. I think it would be good to have it automated.  But it neeeds to be more compact.

A thread with details would be a good start.

Emoji updates to original post would also be good.

> 476 issue-989 SEAL-15000 Udate 3rd party dependencies
:changes-requested:  :test-ok: :merged: 

> 476 issue-989 SEAL-15000 Udate 3rd party dependencies
👍 💯 🍡 

I guess better emojies than my example but you get the point.

**Comment by OBorce** [comment]
:+1: for this

**Comment by sidvenu** [comment]
+1 to this feature request. My organization needs this immensely.

**Comment by moonjoungyoung** [comment]
+1 to this feature. This feature is awesome.

**Comment by Sovianum** [comment]
+1

**Comment by zchbndcc9** [comment]
+1

**Comment by michael0liver** [comment]
+1

**Comment by haigopi** [comment]
+1

**Comment by ivan-nezhura** [comment]
+1

**Comment by katedoctor** [comment]
+ 1

**Comment by nareshkumaryadav** [comment]
+1

**Comment by nusija42** [comment]
+1

**Comment by adri1wald** [comment]
+1

**Comment by alokCrewFoundry** [comment]
Has this been added in the Roadmap?

**Comment by mhuisman** [comment]
+1.  One PR per thread, please.

**Comment by awlevin** [comment]
+1

**Comment by jordie23** [comment]
Is this likely to become a thing?

**Comment by fgarcia44** [comment]
+1

**Comment by rubencodes** [comment]
+1

**Comment by isaak-a** [comment]
+1

**Comment by JenniferBuehler-TomTom** [comment]
+1

**Comment by arnold-almeida** [comment]
This would be a great feature. 

In our team we generally post a PR in the `engineering` channel once a PR is ready for review and we use tags to indicate the status of the PR. 

If PR comments were updated in that thread it would greatly improve the general feedback loop.

**Comment by alexfoxy** [comment]
+1

**Comment by ivorynoise** [comment]
+1

**Comment by kdmgs110** [comment]
+1

**Comment by Zinkutal** [comment]
+1

**Comment by efojs** [comment]
Add `type:feature` to this issue

**Comment by razhollander-monkeyleague** [comment]
+1

**Comment by vovasaz** [comment]
+1

**Comment by isaac-abotsi-mt** [comment]
+1

**Comment by ariefaprilianto** [comment]
It's been three years

**Comment by cosydney** [comment]
Hey, 

for those interested in being able to have Slack conversations that are synced with GitHub we developed [Axolo](https://axolo.co). Axolo creates an ephemeral channel for each pull request that gets archived whenever your merge or close your pull requests on GitHub. In the dedicated channel you can make pull request comment, code comments and get notified of your GitHub checks. 

![threads](https://user-images.githubusercontent.com/10532381/183883586-fd9b76c5-fced-45c3-9b42-2d60a9ca9570.jpg)


This integration helps in making team productive by picking up and reviewing pull requests faster while keeping all your context in a single place: Slack. 

Happy to make a demo if anyone wants, just contact us in the chat on [https://axolo](https://axolo.co)

**Comment by evgi88** [comment]
+1 for the feature

**Comment by gregsdennis** [comment]
I just started getting this in one of my slack workspaces.  Is it possible to turn this off?  I use the integration more as an RSS feed, and it's hard to see where new comments come in.

**Comment by sorah** [comment]
It is impossible to notice updates on issues/PRs if it is posted on thread. threads add nothing usefulness.

**Comment by gregsdennis** [comment]
In several of my workspaces, this change has completely broken me. I have many GH issues and PRs that I'm following that all feed into Slack. But Slack can only have a single thread open at once. Please revert this change or add a way to disable it.

**Comment by ShuzoN** [comment]
> Please revert this change or add a way to disable it.

I agree this idea. Please add  subscribe option "also send to #channel".

**Comment by TheMadBug** [comment]
This feature turned out to probably be the most hated change in this repo's history. (at least for how it applies to issue comment threading).
Here's a feature request to remove it #1500 currently at 382 upvotes. Can we admit that the grand majority of users do not want this, and if made configurable it should not be the default?

I sympathise that a bunch of people requested the feature, and so you did the work and implemented it as asked. THEN a lot more people jumped out of the woodwork and told they hated it. But that's sometimes the risk of feature development.

**Comment by andrewsw** [comment]
Just moseying by to report that between this issue and #1500, we wasted roughly 2 hours of developer time getting our subscriptions reworked into something usable for us. This could have been handled significantly better. 

In the end, via undocumented (AFAICT) `reviews:'channel'` we were able to get it working the way we like, but it didn't need to be this difficult. Opt-in for major interface changes is the way to go, IMO.

Thanks, though, for all the work on this integration.

**Comment by TrevorSayre** [comment]
We found that the threads are handy for colocation of information, but also wanted the thread posts to echo to the channel. So we left threads enabled with `/github settings` and also enabled comments and reviews to be posted to the channel as well as the thread via:

```
/github subscribe org/repo comments:"channel"
/github subscribe org/repo reviews:"channel"
```

Thanks for adding these, although the [README](https://github.com/integrations/slack#broadcast-comments-to-channel) only seems to mention the `comments` option, but the `reviews` option also worked.

**Comment by andrewsw** [comment]
@lyderichti59 check your subscriptions, `/github subscribe list features`, for instances of `comments:"channel"` and `reviews:"channel"`. Those are what cause it to echo to the channel. Replace them with just `comments` and `reviews` to get them threaded. 

If you don't have those `"channel"` items, then you might need to just redo your subscriptions. It seems the migration might leave things in a somewhat inconsistent state. At least, that was our experience.

**Comment by lyderichti59** [comment]
Thanks for your help.
I've listed the features, the comments:"channel" and reviews:"channel" didn't appear.
So, I have unsubscribed to the repo, and re-subscribed to it.
I created a PR and closed it to trigger the behaviour. 
Unfortunately, 'Pull request closed by @lyderichti59' slack message was successfully added to the thread of the PR creation, but it was also echoed in the channel although reviews and comments are not enabled.

I've tried to subscribe to comments and reviews too (explicitely, and without the "channel" part) and reopened + reclosed the previously created PR. It didn't work either : both 'reopen' and 'closed' slack message were re-sent to the channel too.

Thanks for  trying to help. Still stuck so far 🤔

**Comment by Serubin** [comment]
Thanks for posting how to disable this.

While threads may provide context, they also make Slack's "Thread" panel effectively useless for tracking actual conversations. After this change appeared and before I disabled it for my organization, I had 70 new threads that mostly consisted of "PR Merged." All of the threads I was tracking are now effectively lost to the depth of the infinite scroll.

**Comment by taoruicui** [comment]
+1

**Comment by ahmed-hamza-k24** [comment]
> /github subscribe

Did you find any solution for the issue - disable echo message on channel?

**Comment by Mrc0113** [comment]
+1

**Comment by ziwengua** [comment]
+1

**Comment by adrian-85** [comment]
+1

**Comment by mvinayku** [comment]
I have written my own YAML to post the comments to the thread.
<img width="725" alt="image" src="https://github.com/user-attachments/assets/38a2179f-c3bd-415e-a611-5ab3ebe8b732">

**Comment by mvinayku** [comment]
<img width="1257" alt="Screenshot 2024-08-09 at 7 08 07 PM" src="https://github.com/user-attachments/assets/dc43fab0-020c-426d-ba56-cdd980be0076">
 I have created my own YAML file which posts notifications to Slack thread.
